boot/freeldr/freeldr/ntldr/winldr.c

```diff
@@ -499,16 +495,73 @@ LoadModule(
 
 static
 BOOLEAN
-WinLdrIsPaeSupported(IN PCHAR KernelFileName,
+WinLdrIsPaeSupported(IN PLOADER_PARAMETER_BLOCK LoaderBlock,
+                     IN PCSTR BootOptions,
+                     IN USHORT OperatingSystemVersion,
+                     IN PCHAR KernelFileName,
                      IN PCHAR HalFileName)
 {
-    BOOLEAN Result = PaeEnabled;
+    BOOLEAN PaeEnabled = FALSE;
+    BOOLEAN PaeDisabled = FALSE;
+    BOOLEAN NoexecuteDisabled = FALSE;
+    BOOLEAN NoexecuteEnabled = FALSE;
+    BOOLEAN Result;
 
     UNIMPLEMENTED;
 
+    if (OperatingSystemVersion > _WIN32_WINNT_NT4 &&
+        NtLdrGetOption(BootOptions, "PAE"))
+    {
+        /* We found the PAE option. */
+        FIXME("LoadWindowsCore: PAE - TRUE (not implemented)\n");
+        PaeEnabled = TRUE;
+    }
+
+    if (OperatingSystemVersion > _WIN32_WINNT_WIN2K)
+    {
+        if (NtLdrGetOption(BootOptions, "NOPAE"))
+        {
+            /* We found the NOPAE option. */
+            FIXME("LoadWindowsCore: NOPAE - TRUE (not implemented)\n");
+            PaeDisabled = TRUE;
+        }
+
+        if (!LoaderBlock->SetupLdrBlock)
```

Comment:
User's Guide for W2003 PE say [Deployment Tools User Guide (RU) ](http://systemmanager.ru/deploy.ru/index.html?page=h_dep_customize.htm) and translated:
"During Windows Setup, the DEP discovery program does not change the value of the /NoExecute parameter in the Boot.ini file. Because of this, you can move the operating system image, regardless of the DEP processor support on both the host and destination computers."